src/rgw/rgw_rest_s3.cc

```diff
@@ -4477,9 +4477,14 @@ rgw::auth::s3::STSEngine::get_session_token(const boost::string_view& session_to
     ldout(cct, 0) << "ERROR: Decryption failed: " << error << dendl;
     return -EPERM;
   } else {
-    dec_output.append('\0');
-    auto iter = dec_output.cbegin();
-    decode(token, iter);
+    try {
+      dec_output.append('\0');
+      auto iter = dec_output.cbegin();
+      decode(token, iter);
+    } catch (const buffer::error& e) {
+      ldout(cct, 0) << "ERROR: decode SessionToken failed: " << error << dendl;
+      return -ERR_INVALID_IDENTITY_TOKEN;
```

Comment:
This error code is for Invalid Oauth2.0/ openid connect token used in AssumeRoleWithWebIdentity. A new error code can be defined for an Invalid session token or EINVAL can be returned here. Otherwise the code looks good.

Reply:
hi @pritha-srivastava the code have update